arrows/core/transform_detected_object_set.cxx

```diff
@@ -0,0 +1,275 @@
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "transform_detected_object_set.h"
+
+#include <iostream>
+#include <vital/io/camera_io.h>
+#include <vital/config/config_difference.h>
+#include <vital/util/string.h>
+#include <vital/types/bounding_box.h>
+#include <Eigen/Core>
+
+using namespace kwiver::vital;
+
+namespace kwiver {
+namespace arrows {
+namespace core {
+
+// ------------------------------------------------------------------
+transform_detected_object_set::
+transform_detected_object_set()
+  : src_camera_krtd_file_name( "" )
+  , dest_camera_krtd_file_name( "" )
+{
+}
+
+// ------------------------------------------------------------------
+transform_detected_object_set::
+transform_detected_object_set(kwiver::vital::camera_perspective_sptr src_cam,
+			      kwiver::vital::camera_perspective_sptr dest_cam)
+  : src_camera( src_cam )
+  , dest_camera( dest_cam )
+{
+}
+
+// ------------------------------------------------------------------
+vital::config_block_sptr
+transform_detected_object_set::get_configuration() const
+{
+  // Get base config from base class
+  vital::config_block_sptr config = vital::algorithm::get_configuration();
+
+  config->set_value( "src_camera_krtd_file_name", src_camera_krtd_file_name,
+                     "Source camera KRTD file name path" );
+
+  config->set_value( "dest_camera_krtd_file_name", dest_camera_krtd_file_name,
+                     "Destination camera KRTD file name path" );
+
+  return config;
+}
+
+
+// ------------------------------------------------------------------
+void
+transform_detected_object_set::
+set_configuration( vital::config_block_sptr config_in )
+{
+  vital::config_block_sptr config = this->get_configuration();
+
+  config->merge_config( config_in );
+  this->src_camera_krtd_file_name =\
```

Comment:
Are these trailing slashes intended here?  These look like a Pythonic line continuation, but that is not needed in C++.  I'm actually not sure what effect that slash has, but I would remove it here and elsewhere in this function.